iceoryx_binding_c/test/moduletests/test_client.cpp

```diff
@@ -0,0 +1,492 @@
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5ad2c0a6-1f39-44e7-ba7a-c286a1d2d40b");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+
+    void* payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, 32), Eq(AllocationResult_SUCCESS));
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, 32), Eq(AllocationResult_SUCCESS));
+
+    payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, 322), Eq(AllocationResult_RUNNING_OUT_OF_CHUNKS));
+    EXPECT_THAT(payload, Eq(nullptr));
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(2U));
+}
+
+TEST_F(iox_client_test, LoanAlignedFailsWhenNoMoreChunksAreAvailable)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "7720afdd-b106-4081-a79d-0f0edfc1edcb");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+
+    void* payload = nullptr;
+    EXPECT_THAT(iox_client_loan_aligned_request(sut, &payload, 32, 32), Eq(AllocationResult_SUCCESS));
+    EXPECT_THAT(iox_client_loan_aligned_request(sut, &payload, 32, 32), Eq(AllocationResult_SUCCESS));
+
+    payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, 322), Eq(AllocationResult_RUNNING_OUT_OF_CHUNKS));
+    EXPECT_THAT(payload, Eq(nullptr));
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(2U));
+}
+
+TEST_F(iox_client_test, ReleaseWorksOnValidPayload)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "159e9b42-4f43-41df-8449-5891950eb592");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+
+    void* payload = nullptr;
+    iox_client_loan_request(sut, &payload, 32);
+
+    iox_client_release_request(sut, payload);
+
+    EXPECT_THAT(memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(0U));
+}
+
+TEST_F(iox_client_test, LoanAndSendWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "fd83a4cd-6f86-47f8-aa46-e20f34959461");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    connect();
+
+    void* payload = nullptr;
+    EXPECT_THAT(iox_client_loan_request(sut, &payload, sizeof(int64_t)), Eq(AllocationResult_SUCCESS));
+    *static_cast<int64_t*>(payload) = 8912389;
+
+    iox_client_send(sut, payload);
+
+    serverRequestQueue.tryPop()
+        .and_then([&](auto& sharedChunk) {
+            auto msg = static_cast<int64_t*>(sharedChunk.getUserPayload());
+            EXPECT_THAT(*msg, Eq(8912389));
+        })
+        .or_else([&] { GTEST_FAIL() << "Expected request but got none"; });
+}
+
+TEST_F(iox_client_test, ConnectWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4809d1ce-3a53-4981-bded-4709599f62b5");
+    prepareClientInit();
+    iox_client_t sut = iox_client_init(&sutStorage, SERVICE, INSTANCE, EVENT, nullptr);
+    iox_client_connect(sut);
```

Comment:
In order to test this, you need to create a client without the `connectOnCreate` option set